File: ibkr_show_backend/app/schemas/market_sentiment.py
from pydantic import BaseModel


class VixRange(BaseModel):
    label: str
    sentiment: str
    strategy: str
    color: str
    is_current: bool


class FearGreedRange(BaseModel):
    label: str
    sentiment: str
    strategy: str
    color: str
    is_current: bool


class MarketSentimentResponse(BaseModel):
    vix_value: float | None = None
    vix_level: str = ""
    vix_ranges: list[VixRange] = []
    fear_greed_value: int | None = None
    fear_greed_level: str = ""
    fear_greed_ranges: list[FearGreedRange] = []

File: ibkr_show_frontend/src/api/marketSentiment.ts
import { request } from './http'
import { withCache } from '@/utils/cache'

export interface VixRange {
  label: string
  sentiment: string
  strategy: string
  color: string
  is_current: boolean
}

export interface FearGreedRange {
  label: string
  sentiment: string
  strategy: string
  color: string
  is_current: boolean
}

export interface MarketSentimentResponse {
  vix_value: number | null
  vix_level: string
  vix_ranges: VixRange[]
  fear_greed_value: number | null
  fear_greed_level: string
  fear_greed_ranges: FearGreedRange[]
}

export function fetchMarketSentiment(): Promise<MarketSentimentResponse> {
  return withCache(
    'market_sentiment',
    () => request<MarketSentimentResponse>('/api/market-sentiment/')
  )
}
